### ISSUE TITLE
[Request] - preFetch - Allow define plugins to be available

### ISSUE THREAD

**Issue by adamkpurdy** [quasarframework/quasar-cli#170]
This would be nice to have for firebase authentication. Currently for firebase authentication to work with SSR, the firebase package needs to be initialize before the application initializes. Right now the only way to achieve this is through the use of preFetch in the App.vue file.

This has been proven to work, but it would be nice to remove it out of the App.vue and back into a plugin. Unfortunately preFetch is run before the plugin's are loaded.

The idea here would be to allow plugs available via the quasar.config.js in the preFetch prop. So something like
~~~~
preFetch: {
  available: true,
  plugins: ['plugin-1', 'plugin-2']
}
~~~~

**Comment by borutjures** [comment]
Adam suggested to put a link to a working Feathers example where the "startup/prefetch" plugin would be useful: https://github.com/nothingismagick/quasar-starter-ssr-pwa-jest-cypress/pull/2

Feathers initialization is synchronous and must happen before the store is initialized (due to how Feathers-vuex works). This is not possible with plugins which all load after the store. So the workaround is to import the Feathers file/code in the store file. This isn't a problem but you have to put this file somewhere (eg. src/api/feathers.js). It would be cleaner to have it in the plugins folder.

`src/store/index.js`:
https://github.com/nothingismagick/quasar-starter-ssr-pwa-jest-cypress/pull/2/commits/9e52fd151f8afbd197ea0d4a8b621aa1c430695e#diff-e3b1ca4ad5a207c170c97a435b3d1ff3
